packages/resources-provider-plugin/src/index.ts

```diff
@@ -166,7 +151,11 @@ module.exports = (server: ResourceProviderApp): Plugin => {
       )
 
       // initialise resource storage
-      db.init({ settings: config, path: server.config.configPath })
+      const dpa = server.getDataDirPath().split('/')
+      const basePath = dpa.slice(0, dpa.length - 2).join('/')
```

Comment:
Why go up the path and not use `getDataDirPath` as is?

Not sure if this works in Windows.

Reply:
The path has always been based from .signalk
Was using app.config.configPath but in aligning with Server API I moved to using getDataDirPath

Reply:
Sure, but why not align with other plugins?

Reply:
I thought it was aligning as the charts plugin uses a similar approach.

Reply:
The config allows the path to be set relative to .signalk similar to the charts plugin

Reply:
We currently have the following structure under `.signalk`
```
.signalk/
   applicationData/            // webapp settings and data (per webapp)
   plugin-config-data/         // plugin settings and data (per plugin)
   serverstate/                // persisted data that survives a server restart 
   settings.json
   security.json
   baseDeltas.json
```
WebApps and Plugins can persist information for their own purposes but there is no "shared" data area defined.
If a plugin is maintaining information which is more widely of interest, It relies on the knowledge: that the plugin is installed and where to find the data.

Is there merit in defining a location for either: shared data or specific data such as resources?
That way plugins know where to both store and retrieve data.

Reply:
To me resource-provider plugin’s data is private to the plugin. Nothing else should be accessing the files it uses directly. This allows the plugin to evolve how it handles persistence.

We have a clear resources provider api. Let’s imagine I create a resources provider plugin that uses sqlite. Why would its files go anywhere but under plugin-config-data?

If other plugins need access to respurces data we should expose an api for that.

The other files are all managed by server code itself.

Not a huge deal, just don’t (yet?) see motivation to deviate from the normal plugin way.

Charts is a little bit different, because the chart files are not 100% managed by the plugin but the user and there’s even provision to configure that directory explicitly.

Reply:
So then, if I understand correctly, the resources-provider plugin should not offer the option to set the location where files are stored.
This can be removed from the config screen.

Reply:
I’d totally missed that. Yes, i think so.